### PR TITLE
storage: Passphrase dialog fields can reveal the text

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/storaged/crypto-keyslots.jsx [hunk 1]
```diff
@@ -511,20 +510,11 @@ function add_dialog(client, block) {
...
-                          validate: (val, vals) => {
-                              return (vals.new_passphrase.length &&
-                                                        vals.new_passphrase != val &&
-                                                        _("Passphrases do not match"));
-                          }
-                      }),
+            NewPassInput("new_passphrase", _("New passphrase"),
+                         {
+                             visible: vals => !client.features.clevis || vals.type == "luks-passphrase",
+                             validate: val => !val.length && _("Passphrase cannot be empty"),
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:516]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx [hunk 2]
```diff
@@ -558,11 +548,8 @@ function edit_passphrase_dialog(block, key) {
         Fields: [
             PassInput("old_passphrase", _("Old passphrase"),
                       { validate: val => !val.length && _("Passphrase cannot be empty") }),
-            Skip("medskip"),
-            PassInput("new_passphrase", _("New passphrase"),
-                      { validate: val => !val.length && _("Passphrase cannot be empty") }),
-            PassInput("new_passphrase2", _("Repeat passphrase"),
-                      { validate: (val, vals) => vals.new_passphrase.length && vals.new_passphrase != val && _("Passphrases do not match") })
+            NewPassInput("new_passphrase", _("New passphrase"),
+                         { validate: val => !val.length && _("Passphrase cannot be empty") }),
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:552]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 3]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+    const [show, setShow] = useState(false);
+
+    return (
+        <InputGroup>
+            <TextInputPF4 data-field={tag} data-field-type="text-input"
+                          validated={validated}
+                          disabled={options.disabled}
+                          aria-label={title}
+                          autoComplete={autoComplete}
+                          type={show ? "text" : "password"}
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:577]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 4]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+                          value={val}
+                          onChange={(_event, value) => change(value)} />
+            <Button variant="control"
+                    onClick={() => setShow(!show)}
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 5]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+                    isDisabled={options.disabled}>
+                { show ? <EyeSlashIcon /> : <EyeIcon /> }
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:583]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/format-dialog.jsx [hunk 6]
```diff
@@ -323,22 +323,14 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
...
-                                                if (vals.crypto != " keep" && phrase2 != vals.passphrase)
-                                                    return _("Passphrases do not match");
-                                            },
-                                            visible: vals => is_encrypted(vals) && vals.crypto != " keep",
-                                        }),
+                              NewPassInput("passphrase", _("Passphrase"),
+                                           {
+                                               validate: function (phrase, vals) {
+                                                   if (vals.crypto != " keep" && phrase === "")
+                                                       return _("Passphrase cannot be empty");
```
Comment (cockpituous) [inline pkg/storaged/format-dialog.jsx:330]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/stratis-panel.jsx [hunk 7]
```diff
@@ -125,22 +125,14 @@ export function create_stratis_pool(client) {
...
-                                                 if (phrase2 != vals.passphrase)
-                                                     return _("Passphrases do not match");
-                                             },
-                                             visible: vals => vals.encrypt.on
-                                         })
+                               NewPassInput("passphrase", _("Passphrase"),
+                                            {
+                                                validate: function (phrase) {
+                                                    if (phrase === "")
+                                                        return _("Passphrase cannot be empty");
```
Comment (cockpituous) [inline pkg/storaged/stratis-panel.jsx:132]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230608-155257-544e3ca1-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:516]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:552]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:577]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 5 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:583]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/format-dialog.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/format-dialog.jsx:330]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/stratis-panel.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/stratis-panel.jsx:132]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230609-093742-36fdfe4a-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
Pixel diffs: https://github.com/cockpit-project/pixel-test-reference/compare/e85a62d5fb36dddbcedf1c288e1c834b731fdbd4..62513a4c00cfc951b8efabb564e9b023ace9179a

**Comment by martinpitt** [comment]
Is this some new recommendation from PatternFly or somewhere else? This lacks a reference and a justification.

I don't like this. It's IMNSHO very useful to confirm that you can reproduce the passphrase at least once, independent of the revealing. And not let this be done by some Yubikey, a cat walking over your keyboard, or fat-fingering the result.

Also, it's conflicty now, apparently some other pixel update landed in between.

**Review by martinpitt — CHANGES_REQUESTED** [review]
See above

**Comment by mvollmer** [comment]
> Is this some new recommendation from PatternFly or somewhere else? This lacks a reference and a justification.

I thought this is what we are doing in Cockpit, and the Storage page just has to catch up. The sospage uses this pattern (single text entry with reveal functionality for a new password), but I can't find any other... 

@garrett, can you clarify? Should all password fields in Cockpit have a reveal functionality? Should we have a single text input for new passwords, or two?

I know of these articles:
 - https://www.nngroup.com/articles/stop-password-masking/
 - https://www.nngroup.com/articles/password-creation/

**Comment by martinpitt** [comment]
To clarify: I welcome the "reveal" functionality. I don't like the "drop confirmation input" part.

**Comment by garrett** [comment]
Nielsen Norman Group is huge in the UI/UX design space (they've literally written several authoritative books on the subject). Here are several more posts from others about removing the password confirmation:
- https://uxmovement.com/forms/why-the-confirm-password-field-must-die/
- https://www.zuko.io/blog/should-you-use-confirm-password-on-your-forms-and-websites-case-study
- https://cxl.com/blog/password-ux/ (section 4)
   - https://web.archive.org/web/20200227021306/https://www.formulate.com.au/blog/double-entry-form-fields/ (linked from section 4 above; the blog was redone and the post was lost, hence the archive link)
- https://web.dev/sign-in-form-best-practices/#no-double-inputs

And as far as PatternFly is concerned: None of PatternFly's password creation examples have a repeat. But they do (usually) have a reveal.

So: What's the way our users will be creating and storing these passwords?
- Are they copy/pasting from somewhere else?
- Are they typing in what is on a sheet of paper?
- Are they using a password manager (even a browser) to generate the password? 

A confirmation prompt is not needed for any of these examples and actively gets in the way.

Perhaps, if they are making something up themselves it _might_ be useful. But even then, they might type the wrong thing twice. Or have caps lock on. Or number lock (if they're entering keys on the keypad). Or sticky keys might interfere, if they're using accessibility features.

Having a second entry means more busywork. It's also not actually used for confirmation! If we were actually having a second one for confirmation, then we'd actually ask for the password after creation and checking to see if it was unlocked, right? In other words: having a "confirmation" field doesn't actually do anything at all to _validate_ the password to make sure it's correct.

---

Anyway, while it's been the general recommendation to drop the confirmation/repeat/verify password field for around the past decade and a half, it's not commonplace in a lot of OS designs yet.

GNOME current user creation:
Screenshot from 2023-06-13 11-36-08


GNOME next user creation:

add-user
https://gitlab.gnome.org/Teams/Design/settings-mockups/-/blob/master/users/add-user.png

macOS Monterey (previous version) user creation:

Screenshot from 2023-06-13 11-31-08

Windows 11 local user creation:

image

However, these are all user account creation UIs, not disk encryption.

macOS Ventura (latest version) has info here: https://support.apple.com/guide/disk-utility/encrypt-protect-a-storage-device-password-dskutl35612/mac which says, "8. Enter and _verify_ a password, then click Choose." (emphasis added).

Here's GNOME's current UI from GNOME Disks for creating a LUKS volume:

image

(I haven't found anything about it in the various redesigns @ https://gitlab.gnome.org/GNOME/gnome-disk-utility/-/issues/28#note_305987 or https://gitlab.gnome.org/Teams/Design/app-mockups/-/blob/master/disks/disks.png or the recent tracking issue @ https://gitlab.gnome.org/GNOME/gnome-disk-utility/-/issues/274)

So: What has been a "best practice" for the web for well over a decade still hasn't made it into user & disk password creation UI in OSes yet.

Now: I wonder what other PF-using RH products do. (I seem to recall OpenShift not having a repeat, but I couldn't find the screenshot I took a while ago during a demo call.)

**Comment by garrett** [comment]
Summary: While dropping the verify is generally accepted design advice for the web (and there are several articles and studies showing that it is not needed), it's not in practice everywhere (as seen from the OS design screenshots I posted above). And Pitti doesn't agree with it.

We should definitely get the view password in regardless and not block on dropping the repeat password.

File: pkg/storaged/crypto-keyslots.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:516]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:577]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 4 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 5 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:583]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/format-dialog.jsx (hunk 6 above)
Comment (cockpituous) [inline pkg/storaged/format-dialog.jsx:330]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/stratis-panel.jsx (hunk 7 above)
Comment (cockpituous) [inline pkg/storaged/stratis-panel.jsx:132]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-094309-5e721175-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by martinpitt** [comment]
Thanks @garrett for the references and research! To clarify, this wasn't a veto, just a strong feeling that I don't like it.

> Are they copy/pasting from somewhere else?

In that case it's cheap to paste it twice.

> Are they typing in what is on a sheet of paper?

That's exactly the case that I'm concerned about -- if your password is so complicated that you make an accidental typo, it could have major consequences if you don't actually type the password that you intend to. And then after a reboot you can't access your files any more. Proving that the user is able to type it at least twice reduces the chance of that happening.

> Are they using a password manager (even a browser) to generate the password?

Not plausible for an encrypted file system, these are usually unlocked during boot, aren't they? (Or with clevis/tang)

If you feel strongly about dropping it, I don't veto it -- I just wanted to put this into the discussion that I still see value in it.

File: pkg/storaged/crypto-keyslots.jsx [hunk 8]
```diff
@@ -558,11 +548,11 @@ function edit_passphrase_dialog(block, key) {
         Fields: [
             PassInput("old_passphrase", _("Old passphrase"),
                       { validate: val => !val.length && _("Passphrase cannot be empty") }),
-            Skip("medskip"),
             PassInput("new_passphrase", _("New passphrase"),
-                      { validate: val => !val.length && _("Passphrase cannot be empty") }),
-            PassInput("new_passphrase2", _("Repeat passphrase"),
-                      { validate: (val, vals) => vals.new_passphrase.length && vals.new_passphrase != val && _("Passphrases do not match") })
+                      {
+                          validate: val => !val.length && _("Passphrase cannot be empty"),
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:553]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-130730-26582778-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 9]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+            <TextInputPF4 data-field={tag} data-field-type="text-input"
+                          validated={validated}
+                          disabled={options.disabled}
+                          aria-label={title}
+                          autoComplete={options.new_password ? "new-password" : null}
+                          type={show ? "text" : "password"}
+                          value={val}
+                          onChange={(_event, value) => change(value)} />
+            <Button variant="control"
+                    onClick={() => setShow(!show)}
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230613-130730-26582778-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by mvollmer** [comment]
> Therefore, I'm fine with keeping this to passphrase reveal only.

Thanks for figuring this all out! I have updated the PR to only add the reveal functionality.

File: pkg/storaged/crypto-keyslots.jsx [hunk 10]
```diff
@@ -560,9 +562,15 @@ function edit_passphrase_dialog(block, key) {
                       { validate: val => !val.length && _("Passphrase cannot be empty") }),
             Skip("medskip"),
             PassInput("new_passphrase", _("New passphrase"),
-                      { validate: val => !val.length && _("Passphrase cannot be empty") }),
+                      {
+                          validate: val => !val.length && _("Passphrase cannot be empty"),
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:566]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230615-124740-c8218947-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx [hunk 11]
```diff
@@ -560,9 +562,15 @@ function edit_passphrase_dialog(block, key) {
...
+                          new_password: true
+                      }),
             PassInput("new_passphrase2", _("Repeat passphrase"),
-                      { validate: (val, vals) => vals.new_passphrase.length && vals.new_passphrase != val && _("Passphrases do not match") })
+                      {
+                          validate: (val, vals) => vals.new_passphrase.length && vals.new_passphrase != val && _("Passphrases do not match"),
```
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:571]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230615-124740-c8218947-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 12]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+    const [show, setShow] = useState(false);
+
+    return (
+        <InputGroup>
+            <TextInputPF4 data-field={tag} data-field-type="text-input"
+                          validated={validated}
+                          disabled={options.disabled}
+                          aria-label={title}
+                          autoComplete={options.new_password ? "new-password" : null}
+                          type={show ? "text" : "password"}
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:577]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230615-124740-c8218947-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 9 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230615-124740-c8218947-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx [hunk 13]
```diff
@@ -563,6 +564,27 @@ export const TextInput = (tag, title, options) => {
...
+                          value={val}
+                          onChange={(_event, value) => change(value)} />
+            <Button variant="control"
+                    onClick={() => setShow(!show)}
+                    isDisabled={options.disabled}>
+                { show ? <EyeSlashIcon /> : <EyeIcon /> }
```
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:583]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230615-124740-c8218947-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/format-dialog.jsx [hunk 14]
```diff
@@ -330,6 +330,7 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                                                     return _("Passphrase cannot be empty");
                                             },
                                             visible: vals => is_encrypted(vals) && vals.crypto != " keep",
+                                            new_passphrase: true
```
Comment (martinpitt) [inline pkg/storaged/format-dialog.jsx:333]: ITYM `new_password` here, no?

File: pkg/storaged/format-dialog.jsx [hunk 15]
```diff
@@ -338,6 +339,7 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
                                                     return _("Passphrases do not match");
                                             },
                                             visible: vals => is_encrypted(vals) && vals.crypto != " keep",
+                                            new_passphrase: true
```
Comment (martinpitt) [inline pkg/storaged/format-dialog.jsx:342]: dito

File: pkg/storaged/stratis-panel.jsx [hunk 16]
```diff
@@ -131,15 +131,17 @@ export function create_stratis_pool(client) {
                                                  if (phrase === "")
                                                      return _("Passphrase cannot be empty");
                                              },
-                                             visible: vals => vals.encrypt.on
+                                             visible: vals => vals.encrypt.on,
+                                             new_passphrase: true
```
Comment (martinpitt) [inline pkg/storaged/stratis-panel.jsx:135]: same here

File: pkg/storaged/stratis-panel.jsx [hunk 17]
```diff
@@ -131,15 +131,17 @@ export function create_stratis_pool(client) {
...
                                          }),
                                PassInput("passphrase2", _("Confirm"),
                                          {
                                              validate: function (phrase2, vals) {
                                                  if (phrase2 != vals.passphrase)
                                                      return _("Passphrases do not match");
                                              },
-                                             visible: vals => vals.encrypt.on
+                                             visible: vals => vals.encrypt.on,
+                                             new_passphrase: true
```
Comment (martinpitt) [inline pkg/storaged/stratis-panel.jsx:144]: and here

**Review by martinpitt — CHANGES_REQUESTED** [review]
If by some magic `new_passphrase` was intended,then I'd like to understand it. Thanks!

File: pkg/storaged/format-dialog.jsx (hunk 14 above)
Comment (mvollmer) [reply pkg/storaged/format-dialog.jsx]: OMG! Good catch, this should be `new_password`.  Thanks.

File: pkg/storaged/crypto-keyslots.jsx (hunk 10 above)
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:566]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230616-104507-5b7c7f1b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/crypto-keyslots.jsx (hunk 11 above)
Comment (cockpituous) [inline pkg/storaged/crypto-keyslots.jsx:571]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230616-104507-5b7c7f1b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 12 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:577]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230616-104507-5b7c7f1b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 9 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:581]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230616-104507-5b7c7f1b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/dialog.jsx (hunk 13 above)
Comment (cockpituous) [inline pkg/storaged/dialog.jsx:583]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18906-20230616-104507-5b7c7f1b-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by martinpitt — APPROVED** [review]
Danke!

**Review by martinpitt — APPROVED** [review]
Cheers!

**Pull request merged**